### PR TITLE
server: fix defrag bug

### PR DESCRIPTION
**Pull request by WIZARD-CXY**
Now the etcd defrag is opening a new boltb db which use default boltdb options. It will lost the boltdb options like freelistType and initial mmapSize we want to set. In this pr I add them.

### CODE REVIEW

**Comment by ptabor** [comment]
Thank you for spotting this. 
I worry a little bit that with 'next' bolt setting we will have a simillar bug again. 
To be on defensive side we should have: 
  `bolt.getOptions()` 
implemented in bolt that creates struct with options that can be passed to new bolt constructor.

**Comment by WIZARD-CXY** [comment]
@ptabor good point

**Comment by WIZARD-CXY** [comment]
@xiang90 @jingyih @gyuho WDYT

**Comment by ptabor** [comment]
The choice is between: 
  - Backend.GetBoltDBOption()
  - Bolt.GetOptions()
 
I assume in both cases we would store it only 'in memory' as volatile state. 

I think the lower layer it is, the better (and easier to keep in sync).

File: server/mvcc/backend/backend.go [hunk 1]
```diff
@@ -446,6 +455,8 @@ func (b *backend) defrag() error {
 		b.lg.Info(
 			"defragmented",
 			zap.String("path", dbp),
+			zap.String("freelistType", string(options.FreelistType)),
+			zap.Int("initMmapSize", options.InitialMmapSize),
```
Comment (gyuho) [inline server/mvcc/backend/backend.go:459]: Can we make this struct field name consistent?

`init-mmap-size`?

**Comment by WIZARD-CXY** [comment]
> The choice is between:
> 
> * Backend.GetBoltDBOption()
> * Bolt.GetOptions()
> 
> I assume in both cases we would store it only 'in memory' as volatile state.
> 
> I think the lower layer it is, the better (and easier to keep in sync).

I think it is enough to keep in the backend, otherwise we should open new interface `SetBoltdbOption` and `GetBoltdbOption` in bolt

File: server/mvcc/backend/backend.go (hunk 1 above)
Comment (WIZARD-CXY) [reply server/mvcc/backend/backend.go:459]: good point

**Comment by xiang90** [comment]
@ptabor We should fix this bug before the next release?

**Comment by ptabor** [comment]
@xiang90: I think so. What we are missing is second opinion whether we want to make cloning easier in bbolt layer or push it up to GCE.  I think this belongs to bbolt and makeing a new bbolt minor/patch version is not a big deal for us.

**Comment by WIZARD-CXY** [comment]
> @ptabor @WIZARD-CXY Pushing it down to bbolt is cleaner. If we have the bandwidth to do it that we, we shall.

OK， I will make it happen @ptabor @xiang90

File: server/mvcc/backend/backend.go [hunk 2]
```diff
@@ -424,7 +433,7 @@ func (b *backend) defrag() error {
 		b.lg.Fatal("failed to rename tmp database", zap.Error(err))
 	}
 
-	b.db, err = bolt.Open(dbp, 0600, boltOpenOptions)
+	b.db, err = bolt.Open(dbp, 0600, &options)
```
Comment (wpedrak) [inline server/mvcc/backend/backend.go:436]: Isn't replacement of `boltOpenOptions` with `&options` an issue? I specifically mean setting `OpenFile` property 

https://github.com/etcd-io/etcd/blob/c46e96d519a38947188243c859e71ddd733e5065/server/mvcc/backend/backend.go#L384-L386

which was not present in `boltOpenOptions`

**Pull request closed without merging**